### PR TITLE
Ensure compatibility with XGBoost 1.6+ in from_xgboost() method

### PR DESCRIPTION
**Pull request by hcho3**
@trivialfis Would you be able to take a quick look and see if it's reasonable?

### CODE REVIEW

**Comment by hcho3** [comment]
Need to think of a way to test with 1.6 dev version of XGBoost.

File: python/treelite/frontend.py [hunk 1]
```diff
@@ -379,10 +379,15 @@ def from_xgboost(cls, booster):
...
         if not isinstance(booster, xgboost.Booster):
             raise ValueError('booster must be of type `xgboost.Booster`')
-        if parse_version(xgboost.__version__) >= parse_version('1.0.0'):
-            # TODO(hcho3): Remove this line once XGBoost implements a method to save JSON model
-            # in-memory. (Right now, save_model() always saves to the disk.)
-            model_json_str = booster.__getstate__()['handle'].decode('utf-8')
+        xgb_version = parse_version(xgboost.__version__)
+        if xgb_version > parse_version("1.5.2"):
+            # For XGBoost version 1.6.0 and later, use save_raw() to export models as JSON string
+            model_json_str = booster.save_raw(raw_format="json").decode("utf-8")
```
Comment (trivialfis) [inline python/treelite/frontend.py]: Is it necessary to decode it?

**Review by trivialfis — COMMENTED** [review]
That's much better, we are now using API with stable interface.

File: python/treelite/frontend.py (hunk 1 above)
Comment (hcho3) [reply python/treelite/frontend.py]: @trivialfis No it is not necessary. I just need to revise `from_xgboost_json()` to accept a `bytearray` in addition to a string.

**Pull request merged**